openverse_catalog/dags/providers/provider_api_scripts/phylopic.py

```diff
@@ -13,272 +13,263 @@
+        else:
+            # Get all images and limit the results for each request.
+            endpoint = f"{list_endpoint}/{self.offset}/{self.batch_limit}"
+        logger.info(f"Constructed endpoint: {endpoint}")
+        return endpoint
+
+    def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
+        """
+        Since the query range is determined via endpoint, this only increments the range
+        to query.
+        """
+        if self.offset is None:
+            self.offset = 0
+        else:
+            self.offset += self.batch_limit
+        return {}
+
+    def get_should_continue(self, response_json):
+        """
+        Override for upstream "return True". Dated runs will only ever make 1 query so
+        they should not continue to loop.
+        """
+        return not bool(self.date)
+
+    @staticmethod
+    def _get_response_data(response_json) -> dict | list | None:
+        """
+        Intermediate method for pulling out results from a Phylopic API request.
+        """
+        if response_json and response_json.get("success") is True:
+            return response_json.get("result")
+
+    def get_batch_data(self, response_json):
+        """
+        Process the returned IDs.
+
+        The Phylopic API returns only lists of IDs in the initial request. We must take
+        this request and iterate through all the IDs to get the metadata for each one.
+        """
+        data = self._get_response_data(response_json)
+
+        if not data:
+            logger.warning("No content available!")
+            return None
+
+        return data
+
+    @staticmethod
+    def _image_url(_uuid: str) -> str:
+        return f"{PhylopicDataIngester.host}/image/{_uuid}"
+
+    @staticmethod
+    def _get_image_info(
+        result: dict, _uuid: str
+    ) -> tuple[str | None, int | None, int | None]:
+        img_url = None
+        width = None
+        height = None
+
+        image_info = result.get("pngFiles")
+        if image_info:
+            images = list(
+                filter(lambda x: (int(str(x.get("width", "0"))) >= 257), image_info)
+            )
+            if images:
+                image = sorted(images, key=lambda x: x["width"], reverse=True)[0]
+                img_url = image.get("url")
+                if not img_url:
+                    logging.warning(
+                        "Image not detected in url: "
+                        f"{PhylopicDataIngester._image_url(_uuid)}"
+                    )
+                else:
+                    img_url = f"{PhylopicDataIngester.host}{img_url}"
+                    width = image.get("width")
+                    height = image.get("height")
+
+        return img_url, width, height
+
+    @staticmethod
+    def _get_taxa_details(result: dict) -> tuple[list[str] | None, str]:
+        taxa = result.get("taxa", [])
+        # [0].get('canonicalName', {}).get('string')
```

Comment:
Is this comment leftover? If it is still relevant, can you add some explanation for it, please? It is not clear what it's meant to indicate at the moment.

It looks like from the code below it was maybe scratch-pad type stuff, so adding a suggestion to just remove it in case it is no longer relevant.

```suggestion
```

Reply:
Ah, good eye!